cg_lims/EPPs/udf/set/set_barcode.py

```diff
@@ -31,6 +32,14 @@ def get_barcode_set_udf(
     failed_samples = []
 
     for artifact in artifacts:
+        if measurement:
+            if len(artifact.samples) > 1:
+                LOG.info(f"Sample {str(artifact.id)} is a pool, and will be excluded")
+            
+            else:
+                measurement_artifact = artifact
+                artifact = artifact.samples[0].artifact
```

Comment:
Just nu kommer den alltid hämta sample artifact på denna rad. Denna förändring borde göra det mer allmänt.
```suggestion
                artifact = artifact.input_artifact_list()[0]
```